server/sonar-server/src/main/java/org/sonar/server/test/index/TestResultSetIterator.java

```diff
@@ -76,8 +79,16 @@ protected Row read(ResultSet rs) throws SQLException {
     String projectUuid = rs.getString(1);
     String fileUuid = rs.getString(2);
     Date updatedAt = new Date(rs.getLong(3));
-    List<DbFileSources.Test> data = FileSourceDto.decodeTestData(rs.getBinaryStream(4));
-    return toRow(projectUuid, fileUuid, updatedAt, data);
+    List<DbFileSources.Test> tests = Collections.emptyList();
+    InputStream dataInput = rs.getBinaryStream(4);
+    if (dataInput != null) {
+      try {
+        tests = FileSourceDto.decodeTestData(dataInput);
+      } catch (Exception e) {
+        Loggers.get(getClass()).error(String.format("Invalid file_sources.binary_data on row with file_uuid='%s', test file will be ignored", fileUuid), e);
```

Comment:
since we proceed with the indexing even thought the exception occurs, shouldn't this be a warning?